File: src/main/java/com/example/autotest_backend/model/Subject.java
package com.example.autotest_backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "subjects")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Subject {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 10)
    private String inviteCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @OneToMany(mappedBy = "subject", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Topic> topics = new ArrayList<>();
}

File: frontend/src/types/subject.ts
export interface Topic {
  id: number;
  name: string;
}

export interface Subject {
  id: number;
  name: string;
  inviteCode: string;
  topics: Topic[];
}

export interface CreateSubjectRequest {
  name: string;
}

export interface CreateTopicRequest {
  name: string;
}

export interface JoinSubjectRequest {
  inviteCode: string;
}
